src/add-to-project.ts

```diff
@@ -4,7 +4,7 @@ import * as github from '@actions/github'
 // TODO: Ensure this (and the Octokit client) works for non-github.com URLs, as well.
 // https://github.com/orgs|users/<ownerName>/projects/<projectNumber>
 const urlParse =
-  /^(?:https:\/\/)?github\.com\/(?<ownerType>orgs|users)\/(?<ownerName>[^/]+)\/projects\/(?<projectNumber>\d+)/
+  /^(?:https:\/\/)?[^/]+\/(?<ownerType>orgs|users)\/(?<ownerName>[^/]+)\/projects\/(?<projectNumber>\d+)/
```

Comment:
The `(?:https:\/\/)?[^/]+` part of this expression seems like it could be buggy. If the URL protocol is `https://` it will work fine, but if it's anything else (like `http://`), the `[^/]` part will unexpectedly stop at the first forward-slash, which will be in the protocol.

Maybe we shouldn't have a 'beginning-of-string' part of the regex at all? It doesn't seem to really matter what goes before or after the part of the URL we care about. That seems like it would allow us to make a more robust regex that just searches anywhere in the string for a `/orgs/*/projects/*` match:

```regex
\/(?<ownerType>orgs|users)\/(?<ownerName>[^/]+)\/projects\/(?<projectNumber>\d+)
```

Reply:
@iansan5653 Anyway I got an answer that this action not working with GHES. But this change will be needed 100% :) 

https://github.com/actions/add-to-project/issues/214

Reply:
Oh right, I forgot the new Projects hasn't shipped to GHES yet 😄

I'm going to close this PR for now but we can revisit & reopen it once that's shipped